### ISSUE TITLE
Terraform refers to "the plan file".

### ISSUE THREAD

**Issue by coretemp** [hashicorp/terraform#18064]
Terraform v0.11.3 outputs the following unhelpful error message. It should give the full absolute file path to the plan file it is talking about. I am certain that there is another Terraform bug, but it would be easier to report that bug when this is fixed.

```
Failed to load backend: The plan file contains a state with a differing lineage than the current
state. By continuing, your current state would be overwritten by the state
in the plan. Please either update the plan with the latest state or delete
your current state and try again.

"Lineage" is a unique identifier generated only once on the creation of
a new, empty state. If these values differ, it means they were created new
at different times. Therefore, Terraform must assume that they're completely
different states.

The most common cause of seeing this error is using a plan that was
created against a different state. Perhaps the plan is very old and the
state has since been recreated, or perhaps the plan was against a competely
different infrastructure.
```

I believe the underlying issue is related to a Terraform cwd problem, but that's irrelevant here.

**Comment by coretemp** [comment]
@jbardin 
Thanks for your reply. 

I run 
```
terraform plan -state=/home/coretemp/terraform/.generated-prod/terraform.tfstate -out=/home/coretemp/terraform/created_plan-prod .generated-prod
```
and subsequently the associated apply command.

This is the current issue I see: 

```
This plan was saved to: /home/coretemp/src/terraform/created_plan-prod

To perform exactly these actions, run the following command to apply:
    terraform apply "/home/coretemp/src/terraform/created_plan-prod"

Failed to load backend: This plan was created against an older state than is current. Please create
a new plan file against the latest state and try again.

Terraform doesn't allow you to run plans that were created from older
states since it doesn't properly represent the latest changes Terraform
may have made, and can result in unsafe behavior.

Plan Serial:    2
Current Serial: 4
```

The error message can also be improved by providing the date of the "older state" as well as specifying what filename that "older state" had as well as what terraform considers to be "current". I am pretty sure I have followed instructions and did everything to specification. I'd also find it useful if the interface would display the file location of where it gets the "serial" information.

But I don't really see the need for the extra information you requested here; if the extra information I requested is displayed further error reporting and debugging can go from there.

**Comment by coretemp** [comment]
The specific error message above was resolved. Not sure if it will come back, but improved error reporting is always welcome. 

This has low priority for me right now.
